draft-ietf-quic-manageability.md

```diff
@@ -1023,6 +1023,36 @@ not arrive at their destination. Admitting a few packets allows the QUIC
 endpoint to determine that the path accepts QUIC. Sudden drops afterwards will
 result in slow and costly timeouts before abandoning the connection.
 
+## Guiding Path MTU
+
+Some networks are required to be able to carry 1500-byte packets but become
+inefficient at this size, because such packets require fragmentation and
+reassembly across a segment where the effective MTU is smaller (e.g. due to
+IPSEC overhead). This leads to a need to influence a majority of senders to
+use smaller packets, so that the limited reassembly capacity is not exceeded.
+
+If the majority use TCP this can be managed with MSS Clamping, but QUIC
+requires a different approach. Section 14 of {{QUIC-TRANSPORT}} advises senders
+to probe larger sizes using Datagram Packetization Layer PMTU Discovery
+({{?DPLPMTUD=RFC8899}}) or Path Maximum Transmission Unit Discovery
+(PMTUD: {{?RFC1191}} and {{?RFC8201}}). This mechanism will encourage senders to
+approach the maximum size, driving fragmentation that they may not be aware of.
+
+Networks can seek to avoid this by identifying probable QUIC packets
```

Comment:
The problem is that probe packets cannot be identified/distinguished from other QUIc traffic. I believe that's also what section 3.1 (Identifying QUIC traffic) says. If that's no clear we can say more there.

Reply:
My interpretation of quic-transport is that it says:
- Initial packets would not normally exceed 1200 bytes unless the client is aware (from history) that a larger size is supported.
- Endpoints should only send packets larger than this if they are employing either PMTUD or DPLPMTUD.
- All QUIC packets that are not sent in a PMTU probe should be sized to fit within the (currently known) maximum datagram size.

Hence probes should be identifiable as the only packets that are larger than this. Is this wrong?

Reply:
Ian that's interesting. The relevant sentence for me from quic-transport is "Datagrams containing Initial packets MAY exceed 1200 bytes if the sender believes that the network path and peer both support the size that it chooses." In the general case, from where does that belief arise? In starting at 1330 (which I assume is the v6 size) it would work fine on the EE network but it would fail in traversing a network with the IPv6 minimum MTU. Are you relying on successful PMTU discovery from the INITIAL packet to catch that situation?

Reply:
HTTP/3 has to handle fallback to TCP no matter what, and too large a packet and UDP being dropped look identical to a client attempting a handshake.

In this case, the belief that networks support 1330/1350 byte size is based on historical data we gathered showing what MTUs were supported by most networks, so it's not path specific.

Reply:
I believe the recommendation needs to be that path needs to support the minimum QUIC packet size (aka 1200B).  If the path performance is limited when sending larger packets, the devices on-path need to decide consistently send all packets for a flow, or should drop all packets that exceed a specific size. This avoids creating an anomaly that has unpredictable consequences on QUIC protocol mechanisms that probe a path. ((The same predictability is needing when choosing a forwarding path, etc)).

Reply:
Agree that "If the path performance is limited when sending larger packets, the devices on-path need to decide consistently send all packets for a flow, or should drop all packets that exceed a specific size. "
This is however a very limited consistency (one flow). I feel we should go further and ask for consistency for all flows of a particular transport protocol (QUIC in this case). This would permit endpoints to make use of historic knowledge such as "on this network, QUIC only works up to size x".

We should also state the desired goal that it should be consistent for all IP packets, regardless of protocol, but recognise that network may be cautious here, especially when there is a large diversity of protocols.

Reply:
Yes, happy with that.